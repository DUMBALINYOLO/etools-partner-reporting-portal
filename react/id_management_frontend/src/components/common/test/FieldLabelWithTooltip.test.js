import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import FieldLabelWithTooltip from '../FieldLabelWithTooltip';
import TooltipIcon from '../TooltipIcon';

describe('FieldLabelWithTooltip component', () => {
    it('renders properly', () => {
        const children = <div></div>
        const classes = {};
        const infoText = 'help';
        const tooltipIconProps = {};
        const labelProps = {};

        const wrapper = shallow(<FieldLabelWithTooltip
            children={children}
            classes={classes}
            infoText={infoText}
            tooltipIconProps={tooltipIconProps}
            labelProps={labelProps}
        />);

        expect(wrapper.dive().length).toBe(1);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders null if infoText is falsy', () => {
        const newInfoText = '';
        const children = <div></div>
        const classes = {};
        const tooltipIconProps = {};
        const labelProps = {};

        const wrapper = shallow(<FieldLabelWithTooltip
            children={children}
            classes={classes}
            infoText={newInfoText}
            tooltipIconProps={tooltipIconProps}
            labelProps={labelProps}
        />);

        expect(wrapper.contains(<TooltipIcon/>)).toBe(false);
    });
});