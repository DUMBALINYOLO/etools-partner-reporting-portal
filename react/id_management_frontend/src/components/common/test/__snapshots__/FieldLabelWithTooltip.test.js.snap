// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldLabelWithTooltip component renders properly 1`] = `
<FieldLabelWithTooltip
  classes={
    Object {
      "icon": "FieldLabelWithTooltip-icon-3",
      "iconPadding": "FieldLabelWithTooltip-iconPadding-4",
      "label": "FieldLabelWithTooltip-label-2",
      "root": "FieldLabelWithTooltip-root-1",
    }
  }
  infoText="help"
  labelProps={Object {}}
  tooltipIconProps={Object {}}
>
  <div />
</FieldLabelWithTooltip>
`;
